src/Garn/GarnConfig.hs

```diff
@@ -30,9 +31,24 @@ import System.IO.Temp (withSystemTempFile)
 
 -- This needs to be in sync with `DenoOutput` in runner.ts
 data DenoOutput
-  = Success GarnConfig
-  | UserError String
-  deriving (Eq, Show, Generic, FromJSON)
+  = Success Version GarnConfig
+  | UserError Version String
+  deriving stock (Eq, Show, Generic)
+
+instance FromJSON DenoOutput where
+  parseJSON = withObject "DenoOutput" $ \o -> do
+    garnTsLibVersion <- o .: fromString "garnTsLibVersion"
```

Comment:
Technically we're not using the `Version` field in `DenoOutput`. We might in the future, but right now we *only* use this field when parsing the `DenoOutput` fails. (See `OnlyTsLibVersion` below.) But I thought it'd be confusing to *not* add this field here, since it's now required in the `DenoOutput` type in typescript.